### PR TITLE
Use rapture.io to abstract json and ws

### LINKED ISSUE #3: CodacyClient uses Play Json to parse response - creates conflict in projects that use a different version of Play Framework
I have a project using Play 2.5 into which I am trying to integrate sbt-codacy-coverage. Running `abt codacyCoverage` results in the exception below. This is because `codacy-api-scala` depends on Play 2.3 and the `Json` implementation in 2.5 is incompatible with the 2.3 implementation.

The coverage data does, in fact, get uploaded to codacy, and the exception occurs when processing the result. This makes my circle-ci build fail.

java.lang.NoSuchMethodError: play.api.libs.json.JsValue.$bslash(Ljava/lang/String;)Lplay/api/libs/json/JsValue;
    at com.codacy.api.client.CodacyClient.com$codacy$api$client$CodacyClient$$parseJson(CodacyClient.scala:98)
    at com.codacy.api.client.CodacyClient$$anonfun$post$1.apply(CodacyClient.scala:65)
    at com.codacy.api.client.CodacyClient$$anonfun$post$1.apply(CodacyClient.scala:53)
    at com.codacy.api.client.CodacyClient.withWSClient(CodacyClient.scala:108)
    at com.codacy.api.client.CodacyClient.post(CodacyClient.scala:53)
    at com.codacy.api.service.CoverageServices.sendReport(CoverageServices.scala:14)
    at com.codacy.CodacyCoveragePlugin$$anonfun$1.apply(CodacyCoveragePlugin.scala:61)
    at com.codacy.CodacyCoveragePlugin$$anonfun$1.apply(CodacyCoveragePlugin.scala:48)

### PR DESCRIPTION
**Pull request by mrfyda**
Use rapture API to drop play-json and play-ws dependencies.

Fixes #3

### CODE REVIEW

**Comment by mrfyda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- src/main/scala/com/codacy/api/helpers/FileHelper.scala  -1
         

Complexity decreasing per file
==============================
+ src/main/scala/com/codacy/api/helpers/ResultHelper.scala  -1
         

Clones removed
==============
+ src/main/scala/com/codacy/api/client/CodacyClient.scala  -2
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/Codacy/codacy-api-scala/pullRequest?prid=351692&bid=3724319)

File: src/main/scala/com/codacy/api/service/CoverageServices.scala [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+import com.codacy.api.{CoverageFileReport, CoverageReport, Language}
+import rapture.json._
+import rapture.json.formatters.compact._
 
-class CoverageServices(client: CodacyClient) {
+class CoverageServices(client: CodacyClient)(implicit ast: JsonAst) {
+
+  //rafael you are right, there is something wrong with the implicit resultion
+  //as a temporary workaround this helps the compiler
+  //i declared it private so we can remove it easily when rapture fixes the issue
```
Comment (pedrorijo91) [inline src/main/scala/com/codacy/api/service/CoverageServices.scala]: link to the issue?

File: src/main/scala/com/codacy/api/service/CoverageServices.scala [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+  private implicit lazy val ser = implicitly[Serializer[CoverageFileReport,Json]]
```
Comment (pedrorijo91) [inline src/main/scala/com/codacy/api/service/CoverageServices.scala]: link to the issue?

**Comment by mrfyda** [comment]
Codacy Here is an overview of what got changed by this pull request:

``` diff

Coverage decreased per file
===========================
- src/main/scala/com/codacy/api/helpers/FileHelper.scala  -1


Complexity decreasing per file
==============================
+ src/main/scala/com/codacy/api/helpers/ResultHelper.scala  -1


Clones removed
==============
+ src/main/scala/com/codacy/api/client/CodacyClient.scala  -2

```

See the complete overview on [Codacy](https://www.codacy.com/app/Codacy/codacy-api-scala/pullRequest?prid=351692&bid=3724319)

**Review by pedrorijo91 — APPROVED** [review]

**Comment by machadoit** [comment]
@mrfyda @rtfpessoa @johannegger let's merge it? :D

**Pull request merged**